files/sdw-admin.py

```diff
@@ -82,14 +86,167 @@ def copy_config():
 
 def provision_all():
     """
-    Runs provision-all to apply the salt state.highstate on dom0 and all VMs
+    Applies the salt state.highstate on dom0 and all VMs
     """
+    provision_fedora()
+    provision_base_template()
+    provision_vms()
+    configure_logging_vms_early()
+    configure_whonix_customization()
+    configure_vms()
+    sync_appmenus()
+    configure_sys_usb_device_handling()
+    shutdown_to_apply_configurations()
+    print("Provisioning complete. Please reboot to complete the installation.")
+
+
+def run_cmd(args):
+    print(f"Running \"{' '.join(args)}\"")
     try:
-        subprocess.check_call([os.path.join(SCRIPTS_PATH, "scripts/provision-all")])
+        subprocess.check_call(args)
     except subprocess.CalledProcessError:
-        raise SDWAdminException("Error during provision-all")
+        raise SDWAdminException(f"Error while running {' '.join(args)}")
 
-    print("Provisioning complete. Please reboot to complete the installation.")
+
+def qubesctl_call(step_description: str, args: List[str]):
```

Comment:
(You might want a skip_dom0 parameter set to True, that can optionally be overridden, but that fails "safe" and defaults to skipping dom0)